### ISSUE TITLE
How to specify custom functions that can be justified, force-nl'ed

### ISSUE THREAD

**Issue by vemv** [kkinnear/zprint#256]
clojure-mode is able to let users specify custom functions that are able to get "alignment" (in emacs lingo; "justification" for us here)

Good use cases tend to involve anything that receives key-value pairs as varargs.

A real-world example:

```clj
(check! :string table-name
        :int    value)
```

So, I'd be able to tell zprint that whenever `check!` is encountered, `:justify? <val>, :force-nl? <val>, :comma? <val>` can be specified.

(`:comma?` is not crucial, the rest are for applying non-disruptive codebase reformattings)

Is this something zprint is already capable of?

Thanks - V

**Comment by kkinnear** [comment]
Yes, this one is easy.  You already seem to have discovered `:pair {:justify? true}`.  If we assume that is already in your options map, then just making `check!` a `:pair-fn` will do what you want.

```
; Without `:pair {:justify true}`

(czprint i256 {:parse-string? true :fn-map {"check!" :pair-fn}})
(check! :string table-name
        :int value)

; With `:pair {:justify? true}`

(czprint i256 {:parse-string? true :fn-map {"check!" :pair-fn} :pair {:justify? true}})
(check! :string table-name
        :int    value)
```
You can't get commas outside of maps at present.

**Comment by kkinnear** [comment]
Fixed (not that it changed anything) in `1.2.5`
